server/src/test/java/keywhiz/service/providers/AutomationClientAuthFactoryTest.java

```diff
@@ -78,19 +93,12 @@ public void automationClientRejectsClientsWithoutAutomation() {
     factory.provide(request);
   }
 
-  @Test(expected = ForbiddenException.class)
+  @Test(expected = NotAuthorizedException.class)
```

Comment:
shouldn't we still test for ForbiddenException, assuming that the behavior is preserved?

Reply:
This exception did change from `ForbiddenException` to `NotAuthorizedException`. Originally retrieving a `Client` that didn't exist would have caused a `NotAuthorizedException` ([here](https://github.com/square/keywhiz/blob/252383bc5b3b0fbc522d9e6af11f6919cb4c83a9/server/src/main/java/keywhiz/service/providers/ClientAuthFactory.java#L66-L68)) while retrieving an `AutomationClient` that didn't exist threw a `ForbiddenException` ([here](https://github.com/square/keywhiz/blob/252383bc5b3b0fbc522d9e6af11f6919cb4c83a9/server/src/main/java/keywhiz/service/providers/AutomationClientAuthFactory.java#L60-L62)). Both systems threw a `NotAuthorizedException` when client information wasn't provided in the request.

I decided to standardize on `NotAuthorizedException` when there's an error retrieving the client, and `ForbiddenException` when the client is present but not an automation client.